src/frontend/init.js

```diff
@@ -0,0 +1,80 @@
+var Module = {
+  noInitialRun: true,
+  preRun: [],
+  postRun: [],
+  print: (function() {
+    return function(text) {
+      if (arguments.length > 1) text = Array.prototype.slice.call(arguments).join(' ');
+      console.log(text);
+      self.postMessage({
+        'cmd': 'print',
+        'text': text,
+      });
+    }
+  })(),
+  printErr: function(text) {
+    if (arguments.length > 1) text = Array.prototype.slice.call(arguments).join(' ');
+    console.error(text);
+    self.postMessage({
+      'cmd': 'printErr',
+      'text': text,
+    });
+  },
+  setStatus: function(text) {
+    if (!Module.setStatus.last) Module.setStatus.last = { time: Date.now(), text: '' };
+    if (text === Module.setStatus.last.text) return;
+    var m = text.match(/([^(]+)\((\d+(\.\d+)?)\/(\d+)\)/);
```

Comment:
Please add a comment explaining the semantics of this regex.

Reply:
Done.